src/borrow_tracker/mod.rs

```diff
@@ -76,8 +76,10 @@ pub struct FrameState {
 }
 
 impl VisitProvenance for FrameState {
-    fn visit_provenance(&self, _visit: &mut VisitWith<'_>) {
-        // `protected_tags` are already recorded by `GlobalStateInner`.
+    fn visit_provenance(&self, visit: &mut VisitWith<'_>) {
+        for (id, tag) in &self.protected_tags {
```

Comment:
Every tag in here should also be in `GlobalStateInner::protected_tags`. That's why we had the comment.

However it looks like that's only true for the tags, not the Alloc ID. Good catch!
But this should mean we can stop traversing `GlobalStateInner::protected_tags`, since we are now visiting both the tags and alloc IDs here in `FrameState`.

Reply:
Yup, I think I did that?